src/deserialize.rs

```diff
@@ -0,0 +1,69 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
This module moved up to the top level because the serde trait and lifetime stuff makes it a pain to test directly. It's tested indirectly via the `settings` module though, and the underlying `validate` module also has tests.